crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1565,7 +1565,10 @@ where
                             Ok(TransactionInfoResponse {
                                 certified_transaction: Some(inner_certificate),
                                 ..
-                            }) => {
+                            }) if inner_certificate.epoch() == self.committee.epoch  => {
```

Comment:
can we log a warning if this check fails? eventually we should eliminate all cases of this on the validator side.